### PR TITLE
[WIP] Add Colorsensing hardware support

### PR DESCRIPTION
**Pull request by hyperliskdev**
I think this can be improved with a less bright led in/near the sensor. There is no options to affect the brightness of the LED in the code.

### CODE REVIEW

**Review by Keatonion — APPROVED** [review]

**Comment by ewpratten** [comment]
@hyperliskdev There are merge conflicts. Can you fix this?

File: src/main/java/frc/lib5k/components/ColorSensor5k.java [hunk 1]
```diff
@@ -36,7 +37,33 @@ public boolean isReadingEqual(Color8Bit c, double eps) {
...
+        int g = getGreen();
+        int b = getBlue();
+        return new Color8Bit(r, g, b);
+    }
+
+
+    /**
+     * Return the proximity of the sensor. 
+     */
+    public double getProx() {
```
Comment (ewpratten) [inline src/main/java/frc/lib5k/components/ColorSensor5k.java]: I don't see any reason to have this

File: src/main/java/frc/lib5k/components/sensors/ColorSensor.java [hunk 2]
```diff
@@ -0,0 +1,427 @@
+/*
```
Comment (ewpratten) [inline src/main/java/frc/lib5k/components/sensors/ColorSensor.java]: Please don't just copy-paste other libraries into ours. This is why @override exists.

File: src/main/java/frc/robot/RobotConstants.java [hunk 3]
```diff
@@ -167,10 +168,16 @@ public static boolean isCompBot() {
...
 
+        public static final int SPINNER_MOTOR_ID = 5;
+
+        
         /**
-         * Threshold for color comparison
+         * Game Colors
          */
-        public static final double DEFAULT_COLOR_THRESHOLD = 20;
+        public static Color8Bit redGameColor    = new Color8Bit(255, 0, 0);
```
Comment (ewpratten) [inline src/main/java/frc/robot/RobotConstants.java]: Please use proper naming standards for constants (LIKE_THIS)

Also, this should be in fieldconstants, in its own sub-class

File: src/main/java/frc/lib5k/utils/Mathutils.java [hunk 4]
```diff
@@ -92,4 +92,8 @@ public static double map(double value, double input_low, double input_high, doub
         return (value - input_low) * (output_high - output_low) / (input_high - input_low) + output_low;
     }
 
+    // f(x) = log(x/min) / log(max/min)
+    public static double logmap(double value, double min, double max) {
```
Comment (ewpratten) [inline src/main/java/frc/lib5k/utils/Mathutils.java]: This needs a javadoc

**Review by ewpratten — CHANGES_REQUESTED** [review]

**Comment by ewpratten** [comment]
Please add labels for your PR

**Pull request closed without merging**